src/custom/hooks/useRefetchQuoteCallback.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react'
+import { useUpdateQuote, useClearQuote } from 'state/price/hooks'
+import { registerOnWindow } from 'utils/misc'
+import { FeeQuoteParams, getFeeQuote } from 'utils/operator'
+
+/**
+ * @returns callback that fetches a new quote and update the state
+ */
+export function useRefetchQuoteCallback() {
+  const updateQuote = useUpdateQuote()
+  const clearQuote = useClearQuote()
+  registerOnWindow({ updateQuote })
+
+  return useCallback(
+    async ({ sellToken, buyToken, amount, kind, chainId }: FeeQuoteParams) => {
+      // Get a new quote
+      const fee = await getFeeQuote({ chainId, sellToken, buyToken, amount, kind }).catch(err => {
+        console.error('Error fetching the fee', err)
+        return null
+      })
```

Comment:
Why not a try/catch?

Reply:
why try/catch? 

in this case, i just thought it was nicer, i don't increase the cyclomatic complexity, i define a simple function which is reused in both case. A try/catch would rely on having a mutable `let` var, which is not bad, but I felt this is nicer. Not strong opinion though, u might be able to convince me :)

Reply:
Nah, I don't care that much.
Just throwing back at you what you asked me in a similar situation awhile ago =P

Reply:
I don't even remember the case. It would be fun to see what was our conversation and the specific case. You kept it all this time just for striking back! 😃

Anyways, as i said, not strong decision, just a decision, which I hope at least I added some justification to it

Reply:
I never forget ;)

Reply:
🔖✍️